### ISSUE TITLE
Extension issue

### ISSUE THREAD

**Issue by stevenklar** [WyattSL/greyscript#17]
- Issue Type: `Bug`
- Extension Name: `greyscript`
- Extension Version: `2.0.6`
- OS Version: `Darwin arm64 21.3.0`
- VS Code version: `1.65.2`

```json 
{
	"messages": [],
	"activationTimes": {
		"codeLoadingTime": 14,
		"activateCallTime": 0,
		"activateResolvedTime": 4,
		"activationReason": {
			"startup": false,
			"extensionId": {
				"value": "WyattL.greyscript",
				"_lower": "wyattl.greyscript"
			},
			"activationEvent": "onLanguage:greyscript"
		}
	},
	"runtimeErrors": [],
	"runningLocation": 1
}
```

Let me know what I can do for better debug output.

This happened when I wrote: ```pc = ```

Disabled autocomplete, hoverdocs and color picker. Does not help.

``` [2022-03-12 15:38:16.508] [renderer1] [info] UNRESPONSIVE extension host: starting to profile NOW
[2022-03-12 15:38:21.576] [renderer1] [warning] UNRESPONSIVE extension host: 'WyattL.greyscript' took 92% of 5050.25ms, saved PROFILE here: 'file:///var/folders/hl/kmrsyqpj3bs3dm6pjq91fhqh0000gp/T/exthost-d6b0fb.cpuprofile' [{"id":"gc","total":415837,"percentage":8},{"id":"WyattL.greyscript","total":4634205,"percentage":92}]
```

**Comment by ayecue** [comment]
Hey,

thanks for reporting. Might be the type resolve in the autocompleter. Can you share the whole code? Or did you file just contain `pc = ` and nothing else?

**Comment by stevenklar** [comment]
Here it is again when using #import

`[2022-03-30 23:35:40.624] [renderer1] [error] Unexpected string ending at line 1.: Error: Unexpected string ending at line 1.
    at new t (/Users/username/.vscode/extensions/wyattl.greyscript-2.0.6/extension.js:1:47735)
    at t.e.scanStringLiteral (/Users/username/.vscode/extensions/wyattl.greyscript-2.0.6/extension.js:1:57687)
    at t.e.scan (/Users/username/.vscode/extensions/wyattl.greyscript-2.0.6/extension.js:1:54784)
    at t.e.next (/Users/username/.vscode/extensions/wyattl.greyscript-2.0.6/extension.js:1:60814)
    at t.e.prefetch (/Users/username/.vscode/extensions/wyattl.greyscript-2.0.6/extension.js:1:70652)
    at t.e.fetch (/Users/username/.vscode/extensions/wyattl.greyscript-2.0.6/extension.js:1:70493)
    at t.e.next (/Users/username/.vscode/extensions/wyattl.greyscript-2.0.6/extension.js:1:70133)
    at t.e.parseIdentifier (/Users/username/.vscode/extensions/wyattl.greyscript-2.0.6/extension.js:1:71742)
    at t.e.parseRighthandExpression (/Users/username/.vscode/extensions/wyattl.greyscript-2.0.6/extension.js:1:72925)
    at t.e.parseSubExpression (/Users/username/.vscode/extensions/wyattl.greyscript-2.0.6/extension.js:1:77504)
    at t.e.parseExpression (/Users/username/.vscode/extensions/wyattl.greyscript-2.0.6/extension.js:1:78624)
    at t.e.parseCallExpression (/Users/username/.vscode/extensions/wyattl.greyscript-2.0.6/extension.js:1:75209)
    at t.e.parseRighthandExpressionPart (/Users/username/.vscode/extensions/wyattl.greyscript-2.0.6/extension.js:1:74817)
    at t.e.parseRighthandExpressionGreedy (/Users/username/.vscode/extensions/wyattl.greyscript-2.0.6/extension.js:1:72740)
    at t.e.parseAssignmentOrCallStatement (/Users/username/.vscode/extensions/wyattl.greyscript-2.0.6/extension.js:1:82077)
    at t.e.parseStatement (/Users/username/.vscode/extensions/wyattl.greyscript-2.0.6/extension.js:1:84625)
    at t.parseStatement (/Users/username/.vscode/extensions/wyattl.greyscript-2.0.6/extension.js:1:93993)
    at t.e.parseBlock (/Users/username/.vscode/extensions/wyattl.greyscript-2.0.6/extension.js:1:84993)
    at t.parseChunk (/Users/username/.vscode/extensions/wyattl.greyscript-2.0.6/extension.js:1:94122)
    at $e (/Users/username/.vscode/extensions/wyattl.greyscript-2.0.6/extension.js:1:98025)
    at n (/Users/username/.vscode/extensions/wyattl.greyscript-2.0.6/extension.js:3:132374)
    at /Users/username/.vscode/extensions/wyattl.greyscript-2.0.6/extension.js:3:132526
    at u.invoke (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:58:145)
    at v.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:58:1856)
    at R.$acceptModelChanged (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:85:72788)
    at i._doInvokeHandler (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:89:13826)
    at i._invokeHandler (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:89:13510)
    at i._receiveRequest (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:89:12171)
    at i._receiveOneMessage (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:89:10841)
    at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:89:8947
    at u.invoke (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:58:145)
    at v.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:58:1856)
    at d.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:66:19034)
    at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:104:34275
    at u.invoke (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:58:145)
    at v.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:58:1856)
    at d.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:66:19034)
    at o._receiveMessage (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:66:23615)
    at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:66:21149
    at u.invoke (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:58:145)
    at v.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:58:1856)
    at v.acceptChunk (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:66:15865)
    at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:66:14995
    at Socket.P (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:104:13797)
    at Socket.emit (node:events:390:28)
    at addChunk (node:internal/streams/readable:315:12)
    at readableAddChunk (node:internal/streams/readable:289:9)
    at Socket.Readable.push (node:internal/streams/readable:228:10)
    at Pipe.onStreamRead (node:internal/stream_base_commons:199:23)
[2022-03-30 23:36:45.327] [renderer1] [info] UNRESPONSIVE extension host: starting to profile NOW
[2022-03-30 23:36:45.901] [renderer1] [warning] {}
[2022-03-30 23:37:11.519] [renderer1] [error] Cannot call write after a stream was destroyed: Error: Cannot call write after a stream was destroyed
    at __node_internal_captureLargerStackTrace (node:internal/errors:464:5)
    at new NodeError (node:internal/errors:371:5)
    at errorBuffer (node:internal/streams/writable:517:14)
    at onwriteError (node:internal/streams/writable:420:3)
    at onwrite (node:internal/streams/writable:457:7)
    at WriteWrap.onWriteComplete [as oncomplete] (node:internal/stream_base_commons:100:12)
[2022-03-30 23:37:11.547] [renderer1] [error] Extension host (LocalProcess) terminated unexpectedly. No extensions were activated.
[2022-03-30 23:37:11.547] [renderer1] [info] Automatically restarting the extension host.`

**Comment by stevenklar** [comment]
I used import wrong. shame on me.

**Comment by ayecue** [comment]
Yes you are right. From the exception message you shared I assume though that it threw when parsing. I'll take a look during the next few days.

**Comment by ayecue** [comment]
There was an issue with a missing EOF check which then caused an infinite loop. It should be fixed with https://github.com/WyattSL/greyscript/commit/5fd6c86b735d210ca4a64a05c5970e8a2b937951. It will be released in version `2.0.7`.

**Comment by stevenklar** [comment]
Awesome. happy I could help fixing it.
Looking forward to 2.0.7 and get back to greyscripting ;-)
